contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/checkstyle.py

```diff
@@ -227,25 +226,14 @@ def execute(self):
         [vt.target for vt in invalidation_check.invalid_vts]
       )
       for filters, targets in tgts_by_compatibility.items():
-        if self.get_options().interpreter_constraints_whitelist is None and not self._constraints_are_whitelisted(filters):
-          deprecated_conditional(
-            lambda: self.get_options().interpreter_constraints_whitelist is None,
-            '1.14.0.dev2',
-            "Python linting is currently restricted to targets that match the global "
-            "interpreter constraints: {}. Pants detected unacceptable filters: {}. "
-            "Use the `--interpreter-constraints-whitelist` lint option to whitelist "
-            "compatibiltiy constraints."
-            .format(PythonSetup.global_instance().interpreter_constraints, filters)
-          )
-        else:
-          sources = self.calculate_sources([tgt for tgt in targets])
-          if sources:
-            allowed_interpreters = set(interpreter_cache.setup(filters=filters))
-            if not allowed_interpreters:
-              raise TaskError('No valid interpreters found for targets: {}\n(filters: {})'
-                              .format(targets, filters))
-            interpreter = min(allowed_interpreters)
-            failure_count += self.checkstyle(interpreter, sources)
+        sources = self.calculate_sources([tgt for tgt in targets])
```

Comment:
I know this is preexisting, but if you end up pushing another commit to this PR, can you change this to `self.calculate_sources(targets)` ? It's not worth another CI cycle on its own, but if you happen to be making other changes anyway...

Reply:
I saw this, haha, I resisted adjusting it and perhaps I should have just gone ahead. Sadly going to live for now, I am gonna merge this so I can start the publish before a new commit lands